discovery/oci/compute.go

```diff
@@ -0,0 +1,570 @@
+
+	req := identity.GetTenancyRequest{TenancyId: common.String(tenancyOcid)}
+
+	resp, err := d.iamClient.GetTenancy(context.Background(), req)
+	if err != nil {
+		return "", fmt.Errorf("Error when fetching OCI tenancy name for Tenancy %s: %w", tenancyOcid, err)
+	}
+
+	mapFromIDToName := make(map[string]string)
+	mapFromIDToName[tenancyOcid] = "Tenancy(" + *resp.Name + ")" // Tenancy aka root compartment name.
+
+	mapFromIDToParentCmptID := make(map[string]string)
+	mapFromIDToParentCmptID[tenancyOcid] = "" // Since root cmpt does not have a parent.
+
+	var page *string
+	reg := common.StringToRegion(region)
+	d.iamClient.SetRegion(string(reg))
+	for {
+		res, err := d.iamClient.ListCompartments(ctx,
+			identity.ListCompartmentsRequest{
+				CompartmentId:          &tenancyOcid,
+				Page:                   page,
+				AccessLevel:            identity.ListCompartmentsAccessLevelAny,
+				CompartmentIdInSubtree: common.Bool(true),
+			})
+		if err != nil {
+			return "", fmt.Errorf("Could not get list of compartments for tenancy %s: %w", tenancyOcid, err)
+		}
+		for _, compartment := range res.Items {
+			if compartment.LifecycleState == identity.CompartmentLifecycleStateActive {
+				mapFromIDToName[*(compartment.Id)] = *(compartment.Name)
+				mapFromIDToParentCmptID[*(compartment.Id)] = *(compartment.CompartmentId)
+			}
+		}
+		if res.OpcNextPage == nil {
+			break
+		}
+		page = res.OpcNextPage
+	}
+
+	mapFromIDToFullCmptName := make(map[string]string)
+	mapFromIDToFullCmptName[tenancyOcid] = mapFromIDToName[tenancyOcid]
+
+	for len(mapFromIDToFullCmptName) < len(mapFromIDToName) {
+		for cmptID, cmptParentCmptID := range mapFromIDToParentCmptID {
+			_, isCmptNameResolvedFullyAlready := mapFromIDToFullCmptName[cmptID]
+			if !isCmptNameResolvedFullyAlready {
+				if cmptParentCmptID == tenancyOcid {
+					mapFromIDToFullCmptName[cmptID] = mapFromIDToName[tenancyOcid] + "/" + mapFromIDToName[cmptID]
+				} else {
+					fullNameOfParentCmpt, isMyParentNameResolvedFully := mapFromIDToFullCmptName[cmptParentCmptID]
+					if isMyParentNameResolvedFully {
+						mapFromIDToFullCmptName[cmptID] = fullNameOfParentCmpt + "/" + mapFromIDToName[cmptID]
+					}
+				}
+			}
+		}
+	}
+	cmptFullName := mapFromIDToFullCmptName[d.cfg.Compartment]
+	return cmptFullName, nil
+}
+
+func (d *Discovery) instanceMatchesFilterTags(instance core.Instance) bool {
+	totalMatchCount := 0
+
+	for _, tagFilter := range d.cfg.TagFilters {
+		if tagFilter.TagNameSpace != "" {
+			for ns, tags := range instance.DefinedTags {
+				if tagFilter.TagNameSpace == ns {
+					for key, value := range tags {
+						if tagFilter.TagKey == key && tagFilter.TagValue == value {
+							totalMatchCount++
+							break
+						}
+					}
+				}
+			}
+		} else {
+			for key, value := range instance.FreeformTags {
+				if tagFilter.TagKey == key && tagFilter.TagValue == value {
+					totalMatchCount++
+					break
+				}
+			}
+		}
+	}
+	return totalMatchCount == len(d.cfg.TagFilters)
+}
```

Comment:
I would still rather not have an implementation directly in the SD, and then go back later and possibly make a breaking change to switch it over to the API. Is there any reason this can't be done in relabeling anyway?